packages/svelte/src/compiler/phases/1-parse/read/style.js

```diff
@@ -288,7 +290,7 @@ function read_selector(parser, inside_pseudo_class = false) {
 			// nth of matcher must come before combinator matcher to prevent collision else the '+' in '+2n-1' would be parsed as a combinator
 			children.push({
 				type: 'Nth',
-				value: /** @type {string} */ (parser.read(REGEX_NTH_OF)),
+				value: /** @type {string} */ (parser.read(REGEX_NTH_OF)?.trimEnd()), // in case it ends with 'of'
```

Comment:
trimming the value shouldn't be necessary, the value isn't really used afterwards

Reply:
the regex doesn't include whitespaces at the end if it doesn't include `of` anyway, so this should be the only case where it doesn't get trimmed. I edited the test to check for comments, you can merge now if it is all good